### DISCUSSION TITLE
How do i make sure a tool has todays exact date?

### DISCUSSION THREAD

**Discussion by InfluenceLabs** [FlowiseAI/Flowise#2497] [Q&A]
Sorry if this is a nooby question, but I created a custom tool that will book meetings in my calendar based on the user input. The issue is, if the user says something like "book a call for tomorrow", it will book for a random date in 2023, usually around october. 

I asked the AI where it got this date and it says the "system". For context I'm hosting on render.com

Is there any way for me make sure the custom tools finds todays date properly so that if the user says tomorrow, it will actually be tomorrow

**Comment by arekhalpern** [comment]
You could try creating a custom tool to fetch the date and time of the location you are in. I would take a look at the video in the link below for help.

https://docs.flowiseai.com/integrations/langchain/tools/custom-tool

**Reply by mrkrumhausen** [reply]
Super relevant feature that should be built into flowise IMO.

**Reply by harrydu-nis** [reply]
@mrkrumhausen  Totally agree with you.  The chatbot need to be aware of the environment, such as time and location.  This kinds of thing should be a built-in feature of the chatbot.  It must come from client. It should not come from server.  You sever doesn't know your user's location and time. They can be in different time-zone. 

I am proposing adding the environment awareness feature into the FlowiseChatEmbed.   The solution can be additional information in chat history.
